Allow access to stubbed keys via the [] method.

andrewcroome opened the pull request:
When using stubs, resolving `:item` via `container.resolve(:item)` returns the stub. 

The small change included here ensures that `container[:item]` returns the stub as well.

andrewcroome committed:
Allow access to stubbed keys via the [] method.

timriley commented:
I wonder if we could avoid having to add that extra `alias` here by changing our approach in `Dry::Container::Mixin`. Instead of using `alias [] resolve` there (`alias` is lexically scoped, which is why @andrewcroome has had to call it again inside the `Stub` module), we could just make `#[]` a proper method:

``` ruby
def [](key)
  resolve(key)
end
```

This should mean that the `Stub` mixing should work properly with `#[]` without the need for the additional alias, right?

andrewcroome committed:
Make #[] a proper method in Dry::Container::Mixin, rather than an alias.

andrewcroome committed:
Remove aliasing of #[] when stubbing.

This is no longer required as #[] in Dry::Container::Mixin is no longer an alias of #resolve.

andrewcroome commented:
Yes, I think @timriley has the better way. I've amended this with that change.

timriley commented:
@andrewcroome Fantastic, thank you!
